libbeat/outputs/logstash/async.go

```diff
@@ -128,6 +131,18 @@ func (c *asyncClient) publishWindowed(
 	ref *msgRef,
 	data []outputs.Data,
 ) (int, error) {
+	if c.ttl != nil {
+		select {
+		case <-c.ttl:
+			if err := c.Close(); err != nil {
+				return 0, err
+			}
+			if err := c.Connect(0); err != nil {
+				return 0, err
+			}
+		default:
+		}
+	}
```

Comment:
I'm not sure I like to have the reconnect here. reason is the windowing. The output is supposed todo a slow start, increasing the window size to deal with potential timing problems if batch gets too big (e.g. connection lost, as processingt takes too long for very big events/batches). As this PR is supposed to help with load-balancers, after a reconnect the next batch might take another route. Here we should consider resetting the windowing parameters. Maybe move the timeout check before the call to publishWindowed and reset `win` to it's initial parameters.